---
title: "Differential_expression_analysis"
author: "Callum_De_Pasquale"
output: html_document
---
## Overview

This analysis explores gene expression differences in the **GSE183947** RNA-seq dataset, with a focus on breast cancer genes like `BRCA1` and `BRCA2`. Key visualizations include barplots, boxplots, scatterplots, and heatmaps. A Wilcoxon test is used to assess statistical significance in BRCA1 expression between metastatic and non-metastatic samples.

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = "/Users/callumdepasquale/Desktop/demo")
```

```{r, include=FALSE}

# Install required packages if not already installed
packages <- c("dplyr", "tidyverse", "GEOquery", "ggplot2")

install_if_missing <- function(pkg) {
  if (!requireNamespace(pkg, quietly = TRUE)) {
    install.packages(pkg, dependencies = TRUE, repos = "https://cloud.r-project.org")
  }
}

invisible(lapply(packages, install_if_missing))
```

### Load packages
```{r, message=FALSE, warning=FALSE}
library(dplyr)
library(tidyverse)
library(GEOquery)
library(ggplot2)
```


### Read in the data
```{r}
diff_data = read.csv("GSE183947_fpkm.csv")
dim(diff_data)
```

### Get metadata
```{r}
Sys.setenv("VROOM_CONNECTION_SIZE" = 131072 * 1000)
gse = getGEO(GEO = 'GSE183947', GSEMatrix = TRUE )

gse

metadata = pData(phenoData(gse[[1]]))

metadata_modified = metadata %>%
  select(1,10,11,17) %>%
  rename(tissue = characteristics_ch1) %>%
  rename(metastasis = characteristics_ch1.1) %>%
  mutate(tissue = gsub("tissue: ", "", tissue)) %>%
  mutate(metastasis = gsub("metastasis: ", "", metastasis))

# reshaping data 
data_long = diff_data %>%
  rename(gene = X) %>%
  pivot_longer(cols = -gene, names_to = "samples", values_to = "FPKM")

# join dataframes = data_long + metadata_modified

data_long = data_long %>%
  left_join(.,metadata_modified, by = c('samples' = 'description'))
  
# explore data
data_long %>%
  filter(gene == 'BRCA1' | gene == 'BRCA2') %>%
  group_by(gene, tissue) %>%
  summarize(mean_FPKM = mean(FPKM),
            median_FPKM = median(FPKM)) %>%
  arrange(-mean_FPKM)

```

```{r, message = FALSE, warning = FALSE}
# 1. barplot
data_long %>%
  filter(gene == 'BRCA1') %>%
  ggplot(., aes(x = samples, y = FPKM, fill = tissue)) +
  geom_col()

# 2. density
data_long %>%
  filter(gene == 'BRCA1') %>%
  ggplot(., aes(x = FPKM, fill = tissue)) +
  geom_density(alpha = 0.5)

# 3. boxplot
data_long %>%
  filter(gene == 'BRCA1') %>%
  ggplot(., aes(x = metastasis, y = FPKM)) +
  geom_boxplot()

# Wilcoxon test for BRCA1 expression by metastasis status
brca1_data <- data_long %>%
  filter(gene == 'BRCA1')

wilcox.test(FPKM ~ metastasis, data = brca1_data)

# 4. scatterplot
data_long %>%
  filter(gene == 'BRCA1' | gene == 'BRCA2') %>%
  spread(key = gene, value = FPKM) %>%
  ggplot(., aes(x = BRCA1, y = BRCA2, color = tissue)) +
  geom_point() + 
  geom_smooth(method = 'lm', se = FALSE)

# 5. Heatmap
interest_genes = c('BRCA1', 'BRCA2', 'TP53', 'ALK', 'MYCN')

data_long %>%
  filter(gene %in% interest_genes) %>%
  ggplot(., aes(x = samples, y = gene, fill = FPKM)) +
  geom_tile() +
  scale_fill_gradient(low = 'white', high = 'red') +
  theme(axis.text.x = element_text(angle = 270, vjust = 0.5, hjust=1))

```




